src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Minmod.hpp

```diff
@@ -355,10 +362,79 @@ class Minmod<VolumeDim, tmpl::list<Tags...>> {
       return false;
     }
 
+    // Allocate temporary buffer to be used in `limit_one_tensor` where we
+    // otherwise make 1 + 2 * VolumeDim allocations per tensor component for
+    // MUSCL and Pi1, and 1 + 4 * VolumeDim allocations per tensor component for
+    // PiN.
+    size_t total_boundary_points = 0;
+    for (size_t d = 0; d < VolumeDim; ++d) {
+      total_boundary_points += mesh.slice_away(d).number_of_grid_points();
+    }
+    std::unique_ptr<double[], decltype(&free)> temp_buffer(
+        static_cast<double*>(malloc(
+            sizeof(double) * (mesh.number_of_grid_points() +
+                              (VolumeDim > 0 ? total_boundary_points : 0)))),
+        &free);
+    size_t alloc_offset = 0;
+    DataVector u_lin(temp_buffer.get() + alloc_offset,
+                     mesh.number_of_grid_points());
+    alloc_offset += mesh.number_of_grid_points();
+    std::array<DataVector, VolumeDim> temp_boundary_buffer{};
+    for (size_t d = 0; VolumeDim > 0 and d < VolumeDim; ++d) {
+      const size_t num_points = mesh.slice_away(d).number_of_grid_points();
+      temp_boundary_buffer[d].set_data_ref(temp_buffer.get() + alloc_offset,
+                                           num_points);
+      alloc_offset += num_points;
+    }
+    // Compute the slice indices once since this is (surprisingly) expensive
```

Comment:
This computation and the one in the `mean_value` test share a lot of code, and I suspect any future users of `mean_value_on_boundary` will need to reimplement this once again. Could you make it a free function so we can reduce the duplication?

Reply:
I'm not sure. I was vaguely thinking of a function that would compute and return `volume_and_slice_indices` (as declared in line 393 below). Perhaps the user would still have to allocate the `indices_buffer` and pass it into the function... which would be okay, though cleaner if avoidable.

Reply:
Ah, the return type comes in because the caller of the function needs to take ownership of the memory as well as the views into that memory. The unique_ptr holds the memory, the array holds the views. Does that make sense?

Reply:
Ah, so the function would return a pair containing `indices_buffer` and `volume_and_slice_indices`? If so, then yes this is what I was asking for :+1:

Reply:
Yes, that's what I had in mind. I'll factor it out :+1: I'm not sure where to put the function. Right now "MeanValue" seems reasonable, but so would `SliceIterator` or a new header :/

Reply:
Great! To me `SliceIterator` or a new header seems more natural than with `MeanValue`... ¯\\\_(ツ)\_/¯

Reply:
I'd put it in `SliceIterator`

Reply:
what is the status of this?